lib/Doctrine/Record.php

```diff
@@ -185,7 +185,7 @@
      *
      * @var array
      */
-    protected $_invokedSaveHooks = false;
+    protected $_invokedSaveHooks = array();
```

Comment:
Always used as an array here (and in anything that extends), no false checks.  Gets set to an empty array on like 370.